apps/wallet/src/ui/app/pages/home/transfer-coin/index.tsx

```diff
@@ -172,67 +167,98 @@ function TransferCoinPage() {
-            onSubmit={onHandleSubmit}
-        >
-            <StepTwo
-                submitError={sendError}
-                coinSymbol={coinSymbol}
-                coinType={coinType}
-                gasBudgetEstimation={gasBudgetEstimation || null}
-                gasCostEstimation={gasBudgetEstimation || null}
-                gasEstimationLoading={isLoading}
-                onClearSubmitError={handleOnClearSubmitError}
-            />
-        </Formik>
+            <Formik
+                initialValues={formData}
+                validationSchema={validationSchemaStepOne}
+                onSubmit={handleNextStep}
+                enableReinitialize={true}
+            >
+                <SendTokenForm
+                    submitError={sendError}
+                    coinType={coinType}
+                    balance={coinBalance}
+                    gasCostEstimation={gasBudgetEstimation || null}
+                    onClearSubmitError={handleOnClearSubmitError}
+                    onAmountChanged={(anAmount) => setAmountToSend(anAmount)}
+                />
+            </Formik>
+        </>
     );
 
-    const steps = [StepOneForm, StepTwoForm];
-
-    const SendCoin = (
-        <div className={st.container}>
-            <PageTitle
-                title="Send Coins"
-                back={currentStep > 1 ? handleBackStep : '/'}
-            />
-
-            <Content className={st.content}>
-                <Loading loading={loadingBalance}>
-                    <ProgressBar
-                        currentStep={currentStep}
-                        stepsName={['Amount', 'Address']}
-                    />
-                    {steps[currentStep - 1]}
-                </Loading>
+    const StepTwoPreview = (
+        <BottomMenuLayout>
+            <Content>
+                <PreviewTransfer
+                    coinType={coinType}
+                    amount={formData.amount}
+                    to={formData.to}
+                    gasCostEstimation={gasBudgetEstimation || null}
+                    onClearSubmitError={handleOnClearSubmitError}
+                />
             </Content>
-        </div>
+            <Menu
+                stuckClass="sendCoin-cta"
+                className="w-full px-0 pb-0 mx-0 gap-2.5"
+            >
+                <Button
+                    type="button"
+                    variant="secondary"
+                    onClick={() => setCurrentStep(1)}
+                    text={'Back'}
+                    before={<ArrowLeft16 />}
+                />
+
+                <Button
+                    type="button"
+                    variant="primary"
+                    onClick={onHandleSubmit}
+                    size="tall"
+                    text={'Send Now'}
+                    after={<ArrowRight16 />}
+                />
+            </Menu>
+        </BottomMenuLayout>
     );
 
-    return <>{SendCoin}</>;
+    const steps = [StepOne, StepTwoPreview];
```

Comment:
Lets avoid this and just have a boolean like `showPreview && valuesSet` to show preview or SendTokensForm

Reply:
Actually `valuesSet` shouldn't be necessary we should make sure we set `showPreview` to true only when we have all necessary user's input